proposed/tracing.md

```diff
@@ -10,14 +10,103 @@ interfaces and functionality first.
 
-## Definitions
+## 2. Interfaces
 
-* **Framework** - An application framework (or micro-framework) that runs a developers code.  
-  Eg. Laravel, Symfony, CakePHP, Slim
-* **Library** - Any library that developers may include that adds additional functionality  
-  Eg. Image Manipulation, HTTP Clients, 3rd Party SDK's 
-* **Provider** - An organization that offers APM as a service. Typically via a composer package
+## 2.1 Span
+
+See: [AAllport/psr-tracing - SpanInterface.php](https://github.com/AAllport/psr-tracing/blob/main/src/SpanInterface.php)
+
+A span represents a single operation within a trace. A trace is a collection of spans.
+A fully formed span will consist of the following data:
+
+- Name
+- Attributes, an associative array representing key-value pairs
+- Start Time
+- End Time
+- Outcome
+- Exceptions
+
+Upon creation, a new span will not be activated or have any attributes set.
+This allows the user to populate the span with data.
+It is RECOMMENDED that users supply a span with all relevant attributes shortly after creation, notably before
+activation.
+However, Providers SHOULD make every effort to persist attribute data if provided after a span is activated.
+
+To track the time a piece of work takes, a user will call `SpanInterface::activate()`.
+This will set the start time and push the current span into whichever context propagation system the provider chooses.
+Users MAY call `SpanInterface::start()` to populate the span with a start time, but not pushed into context propagation.
+Once complete, the user would call SpanInterface::finish() to set the end-time and “pop” the span from the current
+stack.
+
+Spans MUST support creating a child span via `SpanInterface::createChild(string $spanName): SpanInterface`.
+This SHOULD use `TracerInterface::createSpan()`, then set the subsequent span's parent appropriately.
+
+A typical use would look like the following:
+
+```php
+function imgResize($size=100) {
+    $span = $this->tracer->createSpan('image.resize')
+        ->setAttribute('size',$size)
+        ->activate();
+
+    try{
+      //Resize the image
+      return $resizedImage;
+    
+    } catch (Exception $e) {
+        // Ideally, you would attach the exception to the span here
+        $span->setStatus(SpanInterface::STATUS_ERROR)
+             ->addException($e);
+    } finally {
+        $span->finish();
+    }    
+}
+```
+
+This PSR does not dictate how a span’s internals should be represented, other than it MUST implement the SpanInterface.
+However, it is RECOMMENDED to conform to the [W3C TraceContext specification](https://www.w3.org/TR/trace-context/).
+Providers MUST implement the `SpanInterface::toTraceContextHeaders()` method.
+At a minimum, this should return an empty array.
+Most commonly, providers will return traceParent and baggage headers to pass on to child services.
+
+Providers MUST add the following function signatures to allow data retrieval from their span:
```

Comment:
It is better to provide an interface similar to what other PSRs do instead of defining a set of methods w/o it. See https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-7-http-message.md#3-interfaces

Reply:
Worth noting these methods do exist in the [interface](https://github.com/AAllport/psr-tracing/blob/main/src/SpanInterface.php).
I will leave this comment open until I rephrase it slightly to clarify that the paragraph is drawing attention as opposed to an additional requirement.


> The following methods are aimed at retrieving data from a providers span for programmatic use